zetaclient/chains/ton/readme.md

```diff
@@ -0,0 +1,191 @@
+      "init_state": ""
+    },
+    "message": "ACKDhb/RsdCqGtlos3FdLLfvd1zAoWaVpIKcHDCJEZhVZOzqN6rb6WwR1ddLNXlAdjGA8vupnvWxUsvlfXg9sko="
+  },
+  "out_msgs": [
+    {
+      "@type": "raw.message",
+      "hash": "FM39MwV9b+0kSTGnXtx6e0vf3DLvKSNoSGdl9N44pzw=",
+      "source": "EQCHEV5KAS50fZvOATziJEAQxtXjsPiN28Y0IFGbhhnloNo_",
+      "destination": "EQBICXx9LKsP87t73adonyV3RDHkjHOz7EdhrRJIzHeMKcF4",
+      "value": "10000000",
+      "extra_currencies": [],
+      "fwd_fee": "266669",
+      "ihr_fee": "0",
+      "created_lt": "36905041000002",
+      "body_hash": "lqKW0iTyhcZ77pPDD4owkVfw2qNdxbh+QQt4YwoJz8c=",
+      "msg_data": {
+        "@type": "msg.dataRaw",
+        "body": "te6cckEBAQEAAgAAAEysuc0=",
+        "init_state": ""
+      },
+      "message": ""
+    }
+  ]
+}
+```
+
+### RPC
+
+For more details on the RPC API, 
+see the official documentation: https://docs.ton.org/v3/guidelines/dapps/apis-sdks/api-types
+
+TON provides a binary `lite-server` protocol for data access and transaction submission. 
+The initial integration was implemented using a lite-client, but we eventually encountered infrastructure issues: 
+not many RPC providers support lite-server and know how to operate it.
+
+That's why we switched to the widely adopted toncenter-v2 RPC implementation. Technically, it is a Python webserver 
+with a lite-client wrapper and a JSON API hosted by node providers.
+
+The pruning period differs from provider to provider. Most non-archival nodes prune transactions 
+after ~14 days, which is an important consideration for the observer 
+(specifically for `process_inbound_trackers` and `process_outbound_trackers`)
+
+### Gas
+
+TON has a complex [gas model](https://docs.ton.org/v3/documentation/smart-contracts/transaction-fees/fees) 
+with dynamic pricing. Also, all outbound operations are paid by the **contract**, i.e., the Gateway pays 
+the gas fee for withdrawing to end recipients.
+
+In order to keep gas calculation manageable, we measured the gas cost for all operations empirically and 
+placed a "ceiling" that we treat as the gas fee. This is also a suggested approach by the TON team.
+
+Also, the TON gas price can only be changed via a governance proposal, and all VM operations
+have a predefined gas price, so it should not be an issue. This approach makes transactions slightly "overpriced," but the actual difference in tx cost is typically less than a cent ($0.01).
+
+The Gateway has a getter `int calculate_gas_fee(int op) method_id` for retrieving the transaction gas cost in nanoTON.
+
+See [gateway.fc](https://github.com/zeta-chain/protocol-contracts-ton/blob/main/contracts/gateway.fc#L34).
+
+### Event Logs
+
+There are no event logs in TON like in EVM. For `deposit` and `deposit_and_call`, 
+we need an equivalent mechanism. This is achieved using a feature of TON called an "external message" 
+— an outbound message with no destination. The Gateway wrapper's parsing logic relies on this to detect inbound events.
```

Comment:
It might be useful to have some examples of data extracted when creating theinbound and outbound votes